src/Error/Error.php

```diff
@@ -90,4 +92,31 @@ public function getDiff(): ?string
     {
         return $this->diff;
     }
+
+    /**
+     * @return array{
+     *     type: self::TYPE_*,
+     *     filePath: string,
+     *     source: null|array{message: string, code: int, file: string, line: int},
+     *     appliedFixers: list<string>,
+     *     diff: null|string
+     * }
+     */
+    public function jsonSerialize(): array
+    {
+        return [
+            'type' => $this->type,
+            'filePath' => $this->filePath,
+            'source' => null !== $this->source
```

Comment:
[It's called `source` already](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/3b3bfe90fe8d5d920328a1f36288d2dfacf5d6b3/src/Error/Error.php#L45), so this has nothing to do with this PR. I just added serialisation because these errors are passed as JSON through the socket between the worker and parallelisation operator (main process).